File: src/main/scala/com/dingdo/config/AppConfig.scala
package com.dingdo.config

import com.dingdo.common.util.SpringContextUtil
import org.springframework.beans.factory.annotation.Configurable
import org.springframework.context.ApplicationListener
import org.springframework.context.event.ContextRefreshedEvent

@Configurable
class AppConfig extends ApplicationListener[ContextRefreshedEvent] {

  override def onApplicationEvent(event: ContextRefreshedEvent): Unit = {
    SpringContextUtil.setApplicationContext(event.getApplicationContext)
  }
}

File: src/main/kotlin/com/dingo/config/AppConfig.kt
package com.dingo.config


import cn.hutool.core.io.IoUtil
import com.dingo.common.ObjectMappers
import okhttp3.OkHttpClient
import okhttp3.OkHttpClient.Builder
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import okhttp3.Response
import org.springframework.beans.factory.annotation.Configurable
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.Duration
import kotlin.reflect.KClass

@Configurable
@Configuration
open class AppConfig {

    companion object {
        lateinit var okHttpClient: OkHttpClient
    }

    @Bean
    open fun okHttpClient(): OkHttpClient {
        okHttpClient = Builder()
            .connectTimeout(Duration.ofSeconds(100))
            .readTimeout(Duration.ofSeconds(100))
            .build()
        return okHttpClient
    }
}
